File: LeetCode_3153/Main.cpp
#include<bits/stdc++.h>
using namespace std;

class Solution {
public:
    long long sumDigitDifferences(vector<int>& nums) {
        vector<vector<int>> cnt(11, vector<int> (11, 0));

        long long answer = 0;
        for (int i = 0; i < nums.size(); i++) {
            string s = to_string(nums[i]);
            for (int j = 0; j < s.size(); j++) {
                int c = 0;
                for (int k = 0; k < 10; k++) {
                    c += cnt[j][k];
                }
                c -= cnt[j][s[j] - '0'];
                cnt[j][s[j] - '0'] += 1;
                answer += c;
            }
        }
        return answer;
    }
};

File: LeetCode_3153/Main.py
class Solution:
    def sumDigitDifferences(self, nums):
        cnt = [[0] * 11 for _ in range(11)]
        answer = 0

        for num in nums:
            s = str(num)
            for j in range(len(s)):
                total = sum(cnt[j][:10])     # total previous numbers at position j
                total -= cnt[j][int(s[j])]  # remove same digit count
                
                cnt[j][int(s[j])] += 1
                answer += total

        return answer
